pyam/_aggregate.py

```diff
@@ -181,6 +191,19 @@ def _agg_weight(data, weight, method):
     if not data.droplevel(["variable", "unit"]).index.equals(weight.index):
         raise ValueError("Inconsistent index between variable and weight!")
 
+    if drop_negative is True:
+        if any(i < 0 for i in weight):
```

Comment:
Should be more efficient to do the following:
```suggestion
        if any(weight<0):
```